src/Microsoft.AspNet.Hosting/Program.cs

```diff
@@ -29,7 +29,9 @@ public void Main(string[] args)
             var tempConfig = tempBuilder.Build();
             var configFilePath = tempConfig[ConfigFileKey] ?? HostingJsonFile;
 
-            var appBasePath = _serviceProvider.GetRequiredService<IApplicationEnvironment>().ApplicationBasePath;
+            var appEnv = _serviceProvider.GetRequiredService<IApplicationEnvironment>();
+            appEnv.SetData("CommandLineArguments",args);
```

Comment:
I can see a need for the needing to know about all of the command line arguments supplied via dnx/hosting to be accessible by the web application; with regards to making the command line arguments a bit more first class; I had played with this idea but opted for a simple string[] so the web application could parse the string[] however it liked and not be constrained to any implementation that was put in place.